src/main/java/ats/v1/pkb/ast/AstImpl.java

```diff
@@ -49,9 +49,9 @@ public Node getRoot() {
      */
     @Override
     public Node getNthChild(final Node parent, final int n) {
-        if (n >= parent.getChildren().size()-1)
+        if (n > parent.getChildren().size())
```

Comment:
tutaj tak samo, ciężko patrząc na sam kod domyślić się za co odpowiada "n"

Reply:
Tu akurat się nie zgodzę, biorąc pod uwagę nazwę funkcji (getNthChild) myślę że "n" jest stosowną nazwą parametru.

Reply:
Szczerze powiedziawszy bardziej poszedłbym w stronę czegoś w stylu "number" albo "childNumber", ale w sumie tak też jest ok

Reply:
Biorąc pod uwagę to, że nikt inny raczej nie zajrzy do tego kodu za rok, czy 2, to nie ma znaczenia aż takiego :D